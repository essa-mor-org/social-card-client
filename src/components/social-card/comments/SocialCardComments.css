.Social_comments_input_panel{
    padding-top: var(--Social-space);
    display: flex;
    padding-left: var(--Social-space);
    padding-bottom: var(--Social-space);
}

.Social_comments_image{
    border-radius: 50%;
    width: 32px;
    height: 32px;
    margin-right: 5px;
}

.Social_comments_input{
    border-radius: var(--Social_comment_radius);
    padding: 8px 20px;
    border: 1px solid #ccd0d5;
    font-family: inherit;
    font-size: var(--Social-font-smallmeduim);
    width: var(--Social-comment-width);
    box-sizing: border-box;
    outline: none;
}
