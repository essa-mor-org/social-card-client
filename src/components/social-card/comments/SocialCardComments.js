import React from 'react';
import PropTypes from 'prop-types';
import empty from 'is-empty';

import Comment from './Comment';
import './SocialCardComments.css';

function SocialCardComments({ profileImage, comments }) {
	return (<div>
		<div className="Social_comments_input_panel">
			<img className="Social_comments_image" src={profileImage} alt="" />
			<input className="Social_comments_input" type="text" placeholder="Write a comment..." />
		</div>
		{!empty(comments) && comments.map(({ id, profileImage, profileName, comment }) => 
			<Comment key={id} profileImage={profileImage} profileName={profileName} comment={comment} />)}
	</div>);
}

SocialCardComments.propTypes = {
	profileImage: PropTypes.string.isRequired,
	comments: PropTypes.arrayOf(PropTypes.shape({
		profileImage: PropTypes.string.isRequired,
		profileName: PropTypes.string.isRequired,
		comment: PropTypes.string.isRequired,
		id: PropTypes.number.isRequired
	}))
};

export default SocialCardComments;