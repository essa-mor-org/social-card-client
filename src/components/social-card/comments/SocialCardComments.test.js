import React from 'react';
import SocialCardComments from './SocialCardComments';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('SocialCardComments test', () => {
	it('renders correctly - no comments', () => {
		const tree = renderer
			.create(<SocialCardComments
				onComment={jest.fn()}
				profileImage="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly - with comments', () => {
		const tree = renderer
			.create(<SocialCardComments
				onComment={jest.fn()}
				profileImage="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg"
				comments={[{
					id: 1,
					profileImage: 'https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg',
					profileName: 'Vevo',
					comment: 'Hi'
				}]} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly - add comment and press enter', () => {
		const commentCallBack = jest.fn();
		const wrapper = mount(<SocialCardComments
			onComment={commentCallBack}
			profileImage="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg" />);
		const input = wrapper.find('.Social_comments_input');
		input.simulate('keyPress',{key: 'Enter', target:{value:'abc'}});
		expect(commentCallBack).toHaveBeenCalledWith('abc');
		expect(wrapper.state('comment')).toBe('');
	});

	it('renders correctly - add comment do not press enter', () => {
		const commentCallBack = jest.fn();
		const wrapper = shallow(<SocialCardComments
			onComment={commentCallBack}
			profileImage="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg" />);
		wrapper.find('.Social_comments_input').simulate('keyPress',{key: '', target:{value:'abc'}});
		expect(commentCallBack).toHaveBeenCalledTimes(0);
	});
});
