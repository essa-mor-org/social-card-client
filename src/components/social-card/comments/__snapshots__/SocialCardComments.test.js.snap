// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment test renders correctly - no comments 1`] = `
<div
  className="Comment_panel"
>
  <img
    alt=""
    className="Social_comments_image"
    src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg"
  />
  <div
    className="Comment_comment_panel"
  >
    <a
      className="Comment_profile"
      href="#"
    />
    <span
      className="Comment_comment"
    />
  </div>
</div>
`;

exports[`Comment test renders correctly - with comments 1`] = `
<div
  className="Comment_panel"
>
  <img
    alt=""
    className="Social_comments_image"
    src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg"
  />
  <div
    className="Comment_comment_panel"
  >
    <a
      className="Comment_profile"
      href="#"
    />
    <span
      className="Comment_comment"
    />
  </div>
</div>
`;
